rest/change_point_routes.go

```diff
@@ -35,43 +36,73 @@ func (h *perfGetChangePointsByVersionHandler) Factory() gimlet.RouteHandler {
 	}
 }
 
+const (
+	pageString       string = "page"
+	pageSizeString   string = "pageSize"
+	variantRegex     string = "variantRegex"
+	versionRegex     string = "versionRegex"
+	taskRegex        string = "taskRegex"
+	testRegex        string = "testRegex"
+	measurementRegex string = "measurementRegex"
+)
+
 // Parse fetches the id from the http request.
 func (h *perfGetChangePointsByVersionHandler) Parse(_ context.Context, r *http.Request) error {
 	h.args.ProjectID = gimlet.GetVars(r)["projectID"]
 	vals := r.URL.Query()
 	catcher := grip.NewBasicCatcher()
 	var err error
-	page := vals.Get("page")
+	page := vals.Get(pageString)
+	delete(vals, pageString)
 	if page != "" {
 		h.args.Page, err = strconv.Atoi(page)
 		catcher.Add(err)
 	} else {
 		h.args.Page = 0
 	}
-	pageSize := vals.Get("pageSize")
+	pageSize := vals.Get(pageSizeString)
+	delete(vals, pageSizeString)
 	if pageSize != "" {
 		h.args.PageSize, err = strconv.Atoi(pageSize)
 		catcher.Add(err)
 	} else {
 		h.args.PageSize = 10
 	}
-	h.args.VariantRegex = vals.Get("variantRegex")
-	h.args.VersionRegex = vals.Get("versionRegex")
-	h.args.TaskRegex = vals.Get("taskRegex")
-	h.args.TestRegex = vals.Get("testRegex")
-	h.args.MeasurementRegex = vals.Get("measurementRegex")
-	tls := vals.Get("threadLevels")
-	if tls != "" {
-		tlslice := strings.Split(tls, ",")
-		for _, tl := range tlslice {
-			intTl, err := strconv.Atoi(tl)
-			catcher.Add(err)
-			h.args.ThreadLevels = append(h.args.ThreadLevels, intTl)
+	h.args.VariantRegex = vals.Get(variantRegex)
+	delete(vals, variantRegex)
+	h.args.VersionRegex = vals.Get(versionRegex)
+	delete(vals, versionRegex)
+	h.args.TaskRegex = vals.Get(taskRegex)
+	delete(vals, taskRegex)
+	h.args.TestRegex = vals.Get(testRegex)
+	delete(vals, testRegex)
+	h.args.MeasurementRegex = vals.Get(measurementRegex)
+	delete(vals, measurementRegex)
+	h.args.Arguments = map[string][]int{}
+	for k, v := range vals {
+		key := toSnakeCase(k)
+		val := v[0]
+		if val != "" {
+			valslice := strings.Split(val, ",")
+			for _, valEntry := range valslice {
+				intVal, err := strconv.Atoi(valEntry)
+				catcher.Add(err)
+				h.args.Arguments[key] = append(h.args.Arguments[key], intVal)
+			}
 		}
 	}
 	return catcher.Resolve()
 }
 
+var matchFirstCap = regexp.MustCompile("(.)([A-Z][a-z]+)")
+var matchAllCap = regexp.MustCompile("([a-z0-9])([A-Z])")
```

Comment:
I'd avoid the global variables and just put these in the utility function!

Reply:
(I might also make it return an error if there's a problem parsing the regex or the conversion?)